### DISCUSSION TITLE
Email auth returning null. and not remembering login status.

### DISCUSSION THREAD

**Discussion by Tristannos** [serverpod/serverpod#710] [Q&A] [answered]
I registerd myself into my local DB and i see the record. And when i registerd and enterd the confirmation code i got the userInfo back. But when i reloaded the application for some reason "logged in" is false and when i try to authenticate i see the log in the console that the entry is found:
```
DEBUG: authenticate tristan@gmail.com / XXXXXXXX
DEBUG:  - found entry 
DEBUG:  - 1bc66430fa6856f46b10ee694bf554b779bc3d0f19638b7f4ac0a84fb373a4ee saved: b580c851c452dea419838bc4f21fd1326ca1514f7adcd9b0f516dc70d7c6673f
METHOD CALL: serverpod_auth.email.authenticate duration: 27ms numQueries: 0 authenticatedUser: null
METHOD CALL: serverpod_cloud_storage.file duration: 9ms numQueries: 0 authenticatedUser: null
```
But the function is still returning null, i cant find anymore documentation on this so someone any clue?

**Comment by Tristannos** [comment, marked as answer]
i found out what the problem was.
I implemented authentication and for some reason signup was working fine only login was not.

The easy fix was to run: 
`
dart pub get 
`
`serverpod generate
`

**Comment by eamnicoletti** [comment]
I'm having that problem: when I call the signIn method the `authenticatedUser` got the user id, but right in the next method (different endpoint) the `authenticatedUser` is null.

> METHOD CALL: signin.signIn duration: 1083ms numQueries: 0 authenticatedUser: 1
> METHOD CALL: regionalEndponit.readAll duration: 2ms numQueries: 0 authenticatedUser: null

I've try `dart pub get` and `serverpod generate` , but even so came to nothing.

So, I'm making the authentication logic by myself.

**Reply by Tristannos** [reply]
Did you do the ```dart pub get``` in every folder? 
"client, flutter, server" Then restart the server.
If this doesn't work then maybe try reading the documentation 1 last time step by step.

**Reply by eamnicoletti** [reply]
Yes, I did! But Im still getting the same issue. It's too bad. but I ended up making my own solution for the auth.
